eras/shelley/impl/src/Cardano/Ledger/Shelley/EpochBoundary.hs

```diff
@@ -66,49 +60,12 @@ import Numeric.Natural (Natural)
 
 -- | Type of stake as map from hash key to coins associated.
 newtype Stake crypto = Stake
-  { unStake :: Map (Credential 'Staking crypto) Coin
-  }
+  {unStake :: Map (Credential 'Staking crypto) Coin}
   deriving (Show, Eq, Ord, NoThunks, NFData)
 
-deriving newtype instance
-  CC.Crypto crypto => ToCBOR (Stake crypto)
-
-deriving newtype instance
-  CC.Crypto crypto => FromCBOR (Stake crypto)
-
--- A TxOut has 4 different shapes, depending on the shape its embedded of Addr.
--- Credentials are stored in only 2 of the 4 cases.
--- 1) TxOut (Addr _ _ (StakeRefBase cred)) coin   -> HERE
--- 2) TxOut (Addr _ _ (StakeRefPtr ptr)) coin     -> HERE
--- 3) TxOut (Addr _ _ StakeRefNull) coin          -> NOT HERE
--- 4) TxOut (AddrBootstrap _) coin                -> NOT HERE
--- Unfortunately TxOut is a pattern, that deserializes the address. This can be expensive, so if
--- we only deserialize the parts that we need, for the 2 cases that count, we can speed
--- things up considerably. That is the role of deserialiseAddrStakeRef. It returns (Just stake)
--- for the two cases that matter, and Nothing for the other two cases.
-
--- | Sum up all the Coin for each staking Credential
-aggregateUtxoCoinByCredential ::
-  forall era.
-  ( Era era,
-    HasField "address" (Core.TxOut era) (Addr (Crypto era))
-  ) =>
-  Map Ptr (Credential 'Staking (Crypto era)) ->
-  UTxO era ->
-  Map (Credential 'Staking (Crypto era)) Coin ->
-  Map (Credential 'Staking (Crypto era)) Coin
-aggregateUtxoCoinByCredential ptrs (UTxO u) initial =
-  Map.foldl' accum initial u
```

Comment:
This function has been moved to LedgerState.hs where it has an Incremental analog